state/cleanup.go

```diff
@@ -237,6 +240,66 @@ func (st *State) cleanupRelationSettings(prefix string) error {
 	return nil
 }
 
+func (st *State) cleanupForceDestroyedRelation(prefix string) (err error) {
+	relation, err := st.KeyRelation(prefix)
+	if err != nil {
+		return errors.Annotatef(err, "getting relation %q", prefix)
+	}
+	scopes, closer := st.db().GetCollection(relationScopesC)
+	defer closer()
+
+	sel := bson.D{{"_id", bson.D{{"$regex", "^" + st.docID(relation.globalScope())}}}}
+	iter := scopes.Find(sel).Iter()
+	defer closeIter(iter, &err, "reading relation scopes")
+
+	var doc struct {
+		Key string `bson:"key"`
+	}
+	for iter.Next(&doc) {
+		scope, role, unitName, err := unpackScopeKey(doc.Key)
+		var matchingEp Endpoint
+		for _, ep := range relation.Endpoints() {
+			if string(ep.Role) == role {
+				matchingEp = ep
+			}
+		}
+		if matchingEp.Role == "" {
+			return errors.NotFoundf("endpoint matching %q", doc.Key)
+		}
+
+		// This is nasty but I can't see any other way to do it - we
+		// can't rely on the unit existing to determine the values of
+		// isPrincipal and isLocalUnit, and we're only using the RU to
+		// call LeaveScope on it.
+		ru := RelationUnit{
+			st:          st,
+			relation:    relation,
+			unitName:    unitName,
+			endpoint:    matchingEp,
+			scope:       scope,
+			isPrincipal: true,
+			isLocalUnit: true,
```

Comment:
isLocalUnit is only used on EnterScope() isn't it? So do we even need to set it here?
Same with isPrincipal

Reply:
Well, I thought they might also be used in some other RelationUnit methods - but definitely not in leave scope so no, we don't need to set it here.